### ISSUE TITLE
[Question]: Unsure about a potentional violation

### ISSUE THREAD

**Issue by elizabethsjudd** [IBMa/equal-access#1064]
### Project

a11y checker engine

### Description

I have a basic checkbox group within a fieldset where the checkboxes are disabled. I'm getting a warning about the fact that there are no tab-able elements but isn't that expected if they are disabled? The description for the violation warns about using native HTML elements which I am doing.

Screen shot of the potential violation:
<img width="1553" alt="Screen Shot 2022-08-29 at 6 02 03 PM" src="https://user-images.githubusercontent.com/1275779/187308257-f6762963-0c3c-4365-8846-8f3d16c0f8e3.png">

**Comment by philljenkins** [comment]
Thanks @elizabethsjudd for your question.  
"**_Needs Review_**" items (potential violations) are just for that - the engine is not entirely sure if it is a violation or not, hence the need for human judgement.

Based on your description, there is a basic checkbox group within a fieldset where [ALL] the checkboxes are disabled. Without seeing the code I'm not sure exactly how the screen reader and other AT will behave, but the spec is saying that should be at least one item there that should be tabbable (receive keyboard focus), otherwise how will the user know there is something interactive or not there?   In your example there are two checkboxes there visually, one checked, on not, and so you would need to test with JAWS to determine if it reads to the user the intent of the design.  Similar to the intent of the visual design, that shows one checked and one not, both disabled, the screen reader would need to present the same info to be considered accessible.

Please refer to the resources in the "About this requirement" section in the Help:

**Comment by philljenkins** [comment]
@elizabethsjudd to be able to reproduce (beyond looking of the screen shot) we would need you to provide
1. a link to the test page (or a copy of the test page)
2. any other accessibility bugs reported

**Comment by aliunwala** [comment]
From triage meeting:
@elizabethsjudd Just bumping this to see if you have seen Phill's response.

**Comment by elizabethsjudd** [comment]
Sorry for the delay as I was on vacation. Thank you for the response, I've attached a copy of the page for you to inspect yourself. You are correct that all of the checkboxes are disabled within the fieldset which is why I'm assuming it's having issues. I figured this was valid, while I agree not a great experience, and didn't expect the output of expecting a tab-able element if all tab-able elements are disabled. With that said, I'm not too worry about it as this is simply a demo of our component and I don't believe we have any pages in our product that would have all items disabled like this. I did run through the file with Voice Over and was able to still access the content but as expected could not tab to it since they are disabled.

[components-form-elements--checkbox.html.zip](https://github.com/IBMa/equal-access/files/9568425/components-form-elements--checkbox.html.zip)

**Comment by aliunwala** [comment]
From triage:
Thanks @elizabethsjudd for @shunguoy please take a look if this is still happening in the new rulesets. You mentioned that thus might have been fixed already (fix for disabled checkboxs not to trigger rules)

**Comment by shunguoy** [comment]
@elizabethsjudd  this was fixed during the other fix was applied.  Please check and let us know if you still see the issue.  I will close the issue for now.
